test/unit/query-interface/delete.test.ts

```diff
@@ -39,4 +39,116 @@ describe('QueryInterface#delete', () => {
 
     expect(firstCall.args[1]?.bind).to.be.undefined;
   });
+
+  it('returns fields specified in `options.returning`', async () => {
+    const stub = sinon.stub(sequelize, 'queryRaw');
+    const instance = new User();
+
+    await sequelize.getQueryInterface().delete(
+      instance,
+      User.tableName,
+      { id: ':id' },
+      {
+        returning: [],
+        replacements: {
+          id: '123',
+        },
+      },
+    );
+
+    expect(stub.callCount).to.eq(1);
+    const firstCall = stub.getCall(0);
+    expectsql(firstCall.args[0] as string, {
+      default: 'DELETE FROM `Users` WHERE `id` = \':id\'',
+      postgres: `DELETE FROM "Users" WHERE "id" = ':id' RETURNING *`,
+      mssql: `DELETE FROM [Users] WHERE [id] = N':id'; SELECT @@ROWCOUNT AS AFFECTEDROWS;`,
+      snowflake: `DELETE FROM "Users" WHERE "id" = ':id'`,
```

Comment:
You should be able to delete the special cases of snowflake, db2 and ibmi since the default will handle the differences in the apostrophes etc